core/GlobalState.cc

```diff
@@ -1194,13 +1188,14 @@ void moveNames(pair<unsigned int, unsigned int> *from, pair<unsigned int, unsign
     }
 }
 
-void GlobalState::expandNames(int growBy) {
+void GlobalState::expandNames(u4 newSize) {
     sanityCheck();
-
-    names.reserve(names.capacity() * growBy);
-    vector<pair<unsigned int, unsigned int>> new_namesByHash(namesByHash.capacity() * growBy);
-    moveNames(namesByHash.data(), new_namesByHash.data(), namesByHash.size(), new_namesByHash.capacity());
-    namesByHash.swap(new_namesByHash);
+    if (newSize > names.capacity()) {
+        names.reserve(newSize);
+        vector<pair<unsigned int, unsigned int>> new_namesByHash(newSize * 2);
```

Comment:
```suggestion
        vector<pair<unsigned int, unsigned int>> new_namesByHash(newSize);
```

looking at the prior diff, it looks like `newSize` is pre-multiplied?

Reply:
This is the source of `vector::reserve` in `libc++`:

```cpp
template <class _Tp, class _Allocator>
void
vector<_Tp, _Allocator>::reserve(size_type __n)
{
    if (__n > capacity())
    {
        allocator_type& __a = this->__alloc();
        __split_buffer<value_type, allocator_type&> __v(__n, size(), __a);
        __swap_out_circular_buffer(__v);
    }
}
```

Reply:
https://github.com/sorbet/sorbet/blob/master/core/GlobalState.cc#L1384

`namesByHash` is always 2X the names vector.

The previous version of the code passed in a multiplicative factor, and it increased both by 2X -- keeping the difference in size.